tests/lib.rs

```diff
@@ -209,3 +209,27 @@ fn drop_domain() {
     drop(domain);
     assert_eq!(drops_9001.load(Ordering::SeqCst), 1);
 }
+
+#[non_exhaustive]
+#[derive(Debug)]
+struct Family;
+
+#[test]
+fn hazardptr_compare_exchange_fail() {
+    use haphazard::AtomicPtr as HpAtomicPtr;
```

Comment:
Why alias this instead of just using it as `AtomicPtr`?

Reply:
Because it conflicts with the import of `std`'s `AtomicPtr` for the file: https://github.com/TroyNeubauer/haphazard/blob/fix_compare_exchange_bug/tests/lib.rs#L4

We could stick this in a `mod inner` with its own import of `haphazard`'s `AtomicPtr`, maybe move to a separate file, or leave the alias?

Reply:
My suggestion would be to `use haphazard;` here, then use the full `haphazard::AtomicPtr` path to refer to it. That would make it extra clear, with no need to wonder what the `Hp` in `HpAtomicPtr` stands for.

Reply:
I'm going to be super nitpicky and say that I'd love to see this change in as well. Otherwise I'm happy with this now. Thanks for following up on it!